src/Type/IntegerType.php

```diff
@@ -128,6 +129,22 @@ public function isScalar(): TrinaryLogic
 		return TrinaryLogic::createYes();
 	}
 
+	public function looseCompare(Type $type): BooleanType
+	{
+		foreach ($type->getConstantStrings() as $stringType) {
+			if ($stringType->isNumericString()->yes()) {
+				continue;
+			}
+			return new ConstantBooleanType(false);
```

Comment:
I think this is wrong. So you get a union like `'a'|'123'`, it doesn't have to be false.